# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/forms/nsTextControlFrame.h
# Commit: 2187935f6d78
# Full Hash: 2187935f6d78c22ceeb7e5e92e919aeb42820b95
# Author: David Shin <dshin@mozilla.com>
# Date: 2023-08-16 03:43:43
# Regressor Bug: 1846982
# File Overlap Count: 1
# Description:
#   Bug 1846982: Baseline calculation for textarea. r=emilio
#   
#   Tests are added as tentative, because the behaviour when there's no content,
#   or when there's placeholder text, seem very differnt for each browser.
#   
# ==============================================================================

diff -r 2d475d0406cd -r 2187935f6d78 layout/forms/nsTextControlFrame.h
--- a/layout/forms/nsTextControlFrame.h	Tue Aug 15 13:26:25 2023 +0000
+++ b/layout/forms/nsTextControlFrame.h	Tue Aug 15 13:30:17 2023 +0000
@@ -79,13 +79,10 @@
 
   Maybe<nscoord> GetNaturalBaselineBOffset(
       mozilla::WritingMode aWM, BaselineSharingGroup aBaselineGroup,
-      BaselineExportContext) const override {
-    if (!IsSingleLineTextControl()) {
-      return Nothing{};
-    }
-    NS_ASSERTION(!IsSubtreeDirty(), "frame must not be dirty");
-    return GetSingleLineTextControlBaseline(this, mFirstBaseline, aWM,
-                                            aBaselineGroup);
+      BaselineExportContext aExportContext) const override;
+
+  BaselineSharingGroup GetDefaultBaselineSharingGroup() const override {
+    return BaselineSharingGroup::Last;
   }
 
   static Maybe<nscoord> GetSingleLineTextControlBaseline(